lib/views/request/request_view.dart

```diff
@@ -59,95 +77,66 @@ class _RequestViewState extends State<RequestView> {
-          children: [
-            const Spacer(flex: 1),
-            Expanded(
-              flex: 1,
-              child: Text(
-                display,
-                textScaleFactor: 1.5,
-                style: const TextStyle(
-                  fontSize: 40,
-                  fontWeight: FontWeight.bold,
-                ),
-              ),
-            ),
-            const Spacer(flex: 1),
-            Expanded(
-              child: ButtonBar(
-                buttonHeight: 0,
-                alignment: MainAxisAlignment.spaceEvenly,
-                children: [
-                  numberButton(size, '1'),
-                  numberButton(size, '2'),
-                  numberButton(size, '3')
-                ],
-              ),
-            ),
-            Expanded(
-              child: ButtonBar(
-                buttonHeight: 0,
-                alignment: MainAxisAlignment.spaceEvenly,
-                children: [
-                  numberButton(size, '4'),
-                  numberButton(size, '5'),
-                  numberButton(size, '6')
-                ],
-              ),
-            ),
-            Expanded(
-              child: ButtonBar(
-                buttonHeight: 0,
-                alignment: MainAxisAlignment.spaceEvenly,
-                children: [
-                  numberButton(size, '7'),
-                  numberButton(size, '8'),
-                  numberButton(size, '9')
-                ],
-              ),
-            ),
-            Expanded(
-              child: ButtonBar(
-                buttonHeight: 0,
-                alignment: MainAxisAlignment.spaceEvenly,
-                children: [
-                  numberButton(size, '.'),
-                  numberButton(size, '0'),
-                  numberButton(
-                    size,
-                    '',
-                    icon: Icons.backspace_outlined,
-                  ),
-                ],
+          bottom: true,
+          top: true,
+          child: NumberPadBody(
+            size: MediaQuery.of(context).size,
+            display: display,
+            onNumberButtonTap: (value, icon) {
+              if (icon != null) {
+                if (display.length == 1) {
+                  display = '0';
+                } else {
+                  display = display.substring(0, display.length - 1);
+                }
+              }
+
+              if (display.contains('.')) {
+                if (value == '.') {
+                  return;
+                }
+              }
+
+              if (display.length > 9) {
+                return;
+              }
+
+              setState(() {
+                display = int.parse(display + value).toString();
+              });
+            },
+            onPayButtonPress: () async {
+              if (display == '0') {
+                PopUp.showPopUp(
+                    context, 'Invalid amount', 'Enter an amount > 0', true);
+                return;
+              }
```

Comment:
This PR looks untested, the check is still there

Reply:
No no, actually I added a string at the last of the `request_view`. By clicking on it, you can create an invoice with any amount.

Reply:
Oh no hahah I found it. Why not click on the request button and choose if you want a bitcoin or lightning address?

I think this is much better than finding a tiny label under the button. The user should not know if it is creating an invoice with any amount.